openedx/adg/lms/applications/models.py

```diff
@@ -140,6 +140,15 @@ class Meta:
         verbose_name = _('User Application')
         verbose_name_plural = _('User Applications')
 
+    def save_cover_letter_and_cover_letter_file(self, cover_letter, cover_letter_file):
+        self.cover_letter = cover_letter
+        self.cover_letter_file = cover_letter_file
+        self.save()
```

Comment:
This is misleading. The function name suggests that both cover letter and cover letter file are being saved, while according to the business logic, either cover_lettter or cover_letter_file is saved.

Plus I would advocate against moving this logic to `models.py` @MunamMubashir-arbisoft. The reason is that moving logic to `models.py` makes more sense if we were to override the `save()` method. Like maybe if we wish to do something pre-save or post-save, every time the application is saved.

In this case, bringing this logic at model level is costly performance wise, because two separate calls are made to the save method in `CoverLetterView.post()`. A separate call is made to the db to first save business line, and then a separate call is made to save cover letter. Calls to DB are costly and should be minimized.